site/docs/hosting/cloudflare-workers.md

```diff
@@ -0,0 +1,120 @@
+Specifically, it is a worker which handles HTTP traffic and is written against the [Service Worker API](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).
+Together, with other services that are also provided by Cloudflare, namely:
+
+1. [Workers KV](https://developers.cloudflare.com/workers/runtime-apis/kv/), a global, low-latency, key-value data store,
+2. [Durable Objects](https://developers.cloudflare.com/workers/runtime-apis/durable-objects/), a low-latency coordination and consistent storage for the Workers,
+3. [R2](https://developers.cloudflare.com/r2/), an S3-compatible, zero egress-fee, globally distributed object storage,
+4. [D1](https://developers.cloudflare.com/d1/), a queryable relational database at the edge,
+5. [Queues](https://developers.cloudflare.com/queues/), a message queuing service that allows applications to reliably send and receive messages using Cloudflare Workers.
+
+we can easily create our own Telegram bots, as well as [Web Apps](https://core.telegram.org/bots/webapps), all on the edge, and even for free (with some quotas)!
+
+This guide shows you how you can host your grammY bots on Cloudflare Workers.
+
+## Prerequisites
+
+To follow along, please make sure that you have a Cloudflare account with your workers subdomain configured.
+
+## Setting Things Up
+
+First, create a new project:
+
+```sh
+npx wrangler generate my-bot
+```
+
+You can change `my-bot` to whatever you want.
+This will be the name of your bot and the project directory.
+
+After running the above command, follow the instructions you see to initialize the project.
+There, you can choose between JavaScript or TypeScript.
+
+When the project is initialized, cd into `my-bot` or whatever directory you initialized your project in.
+Depending on how you initialized the project, you should see a file structure similar to the following:
+
+```sh
+.
+|-- node_modules
+|-- package-lock.json
+|-- package.json
+|-- src
+│   └── index.test.js
+│   └── index.js
+`-- wrangler.toml
+```
+
+Next, install `grammy`, and other packages you might need:
+
+```sh
+npm install grammy
+```
+
+## Creating Your Bot
+
+Edit `src/index.js` or `src/index.ts`, and write this code inside:
+
+```js
+// Note that we're importing from 'grammy/web', not 'grammy'.
+import { Bot, webhookCallback } from "grammy/web";
+
+// You can replace `BOT_TOKEN` to your bot token, but it is better to store in an environment variable.
+// For more on this, see https://developers.cloudflare.com/workers/platform/environment-variables/#secrets-on-deployed-workers.
+// You should also replace `BOT_INFO` with your bot info from `bot.api.getMe()`.
+const bot = new Bot(BOT_TOKEN, { botInfo: BOT_INFO });
+
+bot.command("start", async (ctx) => {
+  await ctx.reply("Hello, world!");
+});
+
+addEventListener("fetch", webhookCallback(bot, "cloudflare"));
+```
+
+The above example bot replies "Hello, world!" when it receives the start command.
```

Comment:
```suggestion
The above example bot replies "Hello, world!" when it receives the `/start` command.
```

Reply:
My interpretation:

the start command = `/start`
the `/start` command = the the start command command

What’s yours?

Reply:
Same.

In the end they refer to the same user interaction, and I find `/start` more explicit. Don't you think?

Reply:
"Command" refers to a bot command which refers to a `bot_command` entity which refers to an entity at the beginning of the message starting most of the times with a `/`. Can it be more explicit than this?

Reply:
```suggestion
The above example bot replies "Hello, world!" when it receives `/start`.
```

Reply:
But yes, we can also just leave it. Whatever you prefer :D

Reply:
This one nails it, thanks.